src/town_cmd.cpp

```diff
@@ -2372,19 +2372,25 @@ static Town *CreateRandomTown(uint attempts, uint32_t townnameparts, TownSize si
 	return nullptr;
 }
 
-static const uint8_t _num_initial_towns[4] = {5, 11, 23, 46};  // very low, low, normal, high
-
 /**
  * Generate a number of towns with a given layout.
  * This function is used by the Random Towns button in Scenario Editor as well as in world generation.
  * @param layout The road layout to build.
+ * @param number The number of towns to create, if in the Scenario Editor.
  * @return true if towns have been successfully created.
  */
-bool GenerateTowns(TownLayout layout)
+bool GenerateTowns(TownLayout layout, uint number)
 {
 	uint current_number = 0;
-	uint difficulty = (_game_mode != GM_EDITOR) ? _settings_game.difficulty.number_towns : 0;
-	uint total = (difficulty == (uint)CUSTOM_TOWN_NUMBER_DIFFICULTY) ? _settings_game.game_creation.custom_town_number : Map::ScaleBySize(_num_initial_towns[difficulty] + (Random() & 7));
+	uint total;
+	if (_game_mode == GM_EDITOR) {
+		total = number;
+	} else if (_settings_game.difficulty.number_towns == (uint)CUSTOM_TOWN_NUMBER_DIFFICULTY) {
```

Comment:
Everytime we change code, we try to modernize it:
```suggestion
	} else if (_settings_game.difficulty.number_towns == static_cast<uint>(CUSTOM_TOWN_NUMBER_DIFFICULTY)) {
```